test/performance/io/lowlevel_stream_input_benchmark.cpp

```diff
@@ -0,0 +1,108 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <fstream>
+#include <iostream>
+#include <iterator>
+
+#include <seqan3/alphabet/aminoacid/all.hpp>
+#include <seqan3/io/stream/iterator.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/seqan2.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+
+#if SEQAN3_HAS_SEQAN2
+#include <seqan/stream.h>
+#endif
+
+enum class tag
+{
+    std_stream_it,
+    std_streambuf_it,
+    seqan3_streambuf_it,
+    seqan2_stream_it
+};
+
+template <tag id>
+void consume_all(benchmark::State & state)
```

Comment:
:nail_care: consume_all is a weird name for a benchmark.. maybe `consume_input_stream`, `read_input_stream`, `consume_input` or somehting else, s.t. I can add an analoug output function that tests the ostreambuf iterator